### PR TITLE
Xeno IFF Tags

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Add IFF tags that can be forced into a xeno's carapace to grant them IFF. Xeno handler xenos automatically spawn with these set to Marine/WY IFF groups. A multitool can be used to reprogram the IFF tag. All of these use the IFF group of the ID in your ID slot.

Removed a Wey-Yu crate that doesn't have icons, rearranges the PMC Xeno Handler loadout to be logical and removes an extra r_store item that never got equipped.

Fixed changing IDs not resetting your IFF cache.

## Why It's Good For The Game

Adds new and interesting ways to not shoot your friends.

## Changelog

:cl: Geeves, Nauticall
add: Added xeno IFF tags that prevent smart weapons from shooting the implanted xeno. Use an access tuner to reprogram these, it uses the IFF group on your ID. You can remove it with any surgical tool that has the pinch quality, but remember to be on help intent if you work on xenos.
add: Added the xeno IFF tags to ASRS. Mapped them into Research and the CLs backroom.
add: The PMC Xeno Handler now spawns with an access tuner.
fix: Fixed a bug where smart weapons cached your IFF and didn't bother to set the new IFF if you picked up an ID with a different IFF group.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/damage_procs.dm [hunk 1]
```diff
@@ -1,3 +1,36 @@
+/mob/living/carbon/Xenomorph/attackby(obj/item/I, mob/user)
+	if(user.a_intent == INTENT_HELP)
+		if(HAS_TRAIT(I, TRAIT_TOOL_MULTITOOL) && ishuman(user))
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/damage_procs.dm:3]: Early return here and use a more detailed variable name for I

File: code/modules/mob/living/carbon/xenomorph/items/iff_tag.dm [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	var/option = tgui_alert(programmer, "The xeno tag's current IFF groups reads as: [english_list(faction_groups, "None")]\nYour ID's IFF group reads as: [english_list(id_faction_groups, "None")]", "Xenomorph IFF Tag", list("Overwrite", "Add", "Remove"))
+	if(!option)
+		return FALSE
+	if(xeno)
+		if(!xeno.iff_tag)
+			to_chat(programmer, SPAN_WARNING("\The [src]'s tag got removed while you were reprogramming it!"))
+			return
+		if(!programmer.Adjacent(xeno))
+			to_chat(programmer, SPAN_WARNING("You need to stay close to the xenomorph to reprogram the tag!"))
+			return
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/items/iff_tag.dm:36]: explicitly return false here

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/datums/supply_packs/operations.dm [hunk 3]
```diff
@@ -183,3 +183,13 @@
 	buyable = 0
 	group = "Operations"
 	iteration_needed = null
+
+/datum/supply_packs/xeno_tags
+	name = "xenomorph IFF tag case (x7 tags)"
```
Comment (stanalbatross) [inline code/datums/supply_packs/operations.dm:188]:
```suggestion
	name = "Xenomorph IFF Tag Case (x7 tags)"
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/items/iff_tag.dm [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+/obj/item/iff_tag/attack(mob/living/carbon/Xenomorph/xeno, mob/living/carbon/human/injector)
+	if(isXeno(xeno))
+		injector.visible_message(SPAN_NOTICE("[injector] starts forcing \the [src] into [xeno]'s carapace..."), SPAN_NOTICE("You start forcing \the [src] into [xeno]'s carapace..."))
+		if(!do_after(injector, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC, xeno, INTERRUPT_DIFF_LOC, BUSY_ICON_GENERIC))
+			return
+		injector.visible_message(SPAN_NOTICE("[injector] forces \the [src] into [xeno]'s carapace!"), SPAN_NOTICE("You force \the [src] into [xeno]'s carapace!"))
+		xeno.iff_tag = src
+		injector.drop_inv_item_to_loc(src, xeno)
+		return
+	return ..()
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/items/iff_tag.dm:17]: should not be able to put more than one tag into a xeno as it causes large issues + should also not be able to put them in dead xenos

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**